core/description/resource_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package description
+
+import (
+	"time"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	"gopkg.in/yaml.v2"
+)
+
+type ResourceSuite struct {
+	SliceSerializationSuite
+}
+
+var _ = gc.Suite(&ResourceSuite{})
+
+func (s *ResourceSuite) SetUpTest(c *gc.C) {
+	s.SerializationSuite.SetUpTest(c)
+	s.importName = "resources"
+	s.sliceName = "resources"
+	s.importFunc = func(m map[string]interface{}) (interface{}, error) {
+		return importResources(m)
+	}
+	s.testFields = func(m map[string]interface{}) {
+		m["resources"] = []interface{}{}
+	}
+}
+
+func minimalResourceMap() map[interface{}]interface{} {
```

Comment:
Close enough :)  One revision and one charm store revision is fairly small and exercises most of the functionality.